src/mme-app/actionHandlers/attachActionHandlers.cpp

```diff
@@ -38,11 +54,99 @@
 #include <utils/mmeCauseUtils.h>
 #include "gtpCauseTypes.h"
 
+
+
+#define MCC_MNC_LEN 4
+#define LB_HB_LEN 8
+
 using namespace SM;
 using namespace mme;
 using namespace cmn;
 using namespace cmn::utils;
 
+cmn::ipc::IpcAddress destAddr_dns;
+struct CS_Q_msg cs_msg;
+extern mme_config g_mme_cfg;
+extern MmeIpcInterface* mmeIpcIf_g;
+Void SetDNSConfiguration()
+{
+	DNS::Cache::setRefreshConcurrent( g_mme_cfg.dns_config.concurrent);
+	DNS::Cache::setRefreshPercent( g_mme_cfg.dns_config.percentage);
+	DNS::Cache::setRefreshInterval( g_mme_cfg.dns_config.interval_seconds );
+	DNS::Cache::getInstance().addNamedServer(g_mme_cfg.dns_config.dns1_ip);
+	DNS::Cache::getInstance().applyNamedServers();
+
+}
+
+	template <class T>
+std::string hexFormatWithoutCommas(T value, int width = sizeof(T)*2)
+{
+	struct Numpunct : public std::numpunct<char>
+	{
+		protected:
+			virtual char do_thousands_sep() const { return ' '; }
+			virtual std::string do_grouping() const { return ""; }
+	};
+	std::stringstream ss;
+	ss.imbue({std::locale(), new Numpunct});
+	ss << std::setfill('0') << std::setw(width) << std::hex << value;
+	return ss.str();
+}
+
+
+
+void Ipv4_gateway_ip( EPCDNS::StringVector &ipv4_ip, unsigned long * sgw_ip)
+{
+	EPCDNS::StringVector::const_iterator it = ipv4_ip.begin();
+	std::cout<<*it <<"\n";
```

Comment:
Why don't we iterate through all the ips? in which case would we use the other ips.